documentation/External-Grafana/ReadMe.md

```diff
@@ -0,0 +1,179 @@
+```
+
+> CRD design for `GrafanaInstance` with a default setup for an external Grafana Instance. With this all Grafana Operator operations such as `GrafanaDataSource`, `GrafanaDashboard` will point to this default instance :
+
+```.yaml
+apiVersion: integreatly.org/v1alpha1
+kind: GrafanaInstance
+metadata:
+  name: amazon-managed-grafana
+  annotations:
+    grafanainstance.kubernetes.io/is-default-grafana: "true"
+spec:
+  external:
+    url: <type ConfigMapSelector>
+    grafana_api_key: <type SecretKeySelector>
+```
+
+### Creating Grafana dashboards on Amazon Managed Grafana.
+
+Today Grafana operator supports the creation of Grafana dashboards on self managed grafana instance in the cluster where the Grafana operator is installed. Grafana operator should be enhanced to support creating dashboards on external Grafana instance. 
+
+> In order to accomplish this,  `GrafanaDashboard` CRD should be modified to use `GrafanaInstance` as an input parameter to point to external Grafana instance. This allows customers to use GitOps approach to create Grafana Dashboards on remote Grafana instances.
+
+```.yaml
+apiVersion: integreatly.org/v1alpha1
+kind: GrafanaDashboard
+metadata: 
+  name: grafana-dashboard-from-url
+  labels: 
+    app: grafana
+spec:
+  grafanainstance: "amazon-managed-grafana"
+  url: https://raw.githubusercontent.com/integr8ly/grafana-operator/master/deploy/examples/remote/grafana-dashboard.json
+```
+
+### Adding data sources to external Grafana instance.
+
+Today Grafana operator supports [Amazon Managed Service for Prometheus](https://github.com/grafana-operator/grafana-operator/blob/master/deploy/examples/datasources/AWS-Prometheus.yaml) as a data source. Grafana operator solution should be enhanced to add following data sources (minimally Amazon CloudWatch) :
+- Amazon CloudWatch
+- Amazon OpenSearch Service
+- AWS IoT SiteWise
+- AWS IoT TwinMaker
+- Amazon Timestream
+- Amazon Athena
+- Amazon Redshift
+- AWS X-Ray
+
+One example per below for each of the above listed data source in `grafana-operator` github is desirable for our users.
+
+> CRD design for `GrafanaInstance` to add external cloud based data sources for an external Grafana Instance. With this all Grafana Operator operations such as `GrafanaDataSource`, `GrafanaDashboard` will point to this default instance :
+
+```.yaml
+apiVersion: integreatly.org/v1alpha1
+kind: GrafanaDataSource
+metadata:
+  name: aws-cloudwatch-datasource
+spec:
+  grafanainstance: "amazon-managed-grafana"
+  datasources:
+  - access: proxy
+    editable: true
+    isDefault: true
+    name: cloudwatch
+    type: cloudwatch
+    region: <type ConfigMapSelector>
+    iamRole: <type ConfigMapSelector>
+```
+
+Note: `grafanainstance` is not mandatory if its setup as default external Grafana instance in `GrafanaInstance` CRD. 
+
+This is not a road block feature but nice to have to support all cloud based data sources.
+
+### Creating Grafana alerting on Amazon Managed Grafana.
```

Comment:
To get smaller scopes I would really like to have GrafanaAlerting CRD in a separate PR.
It will make it easier to discuss how we should solve things and also easier to close specific tasks.

Reply:
@NissesSenap I will work on a new Design doc PR on this as soon as possible. I agree we can decouple alerting from this PR at this point of time.